pkg/generator.go

```diff
@@ -472,19 +473,19 @@ func (g *Generator) renderType(ctx context.Context, typ types.Type) string {
 		case 0:
 			return fmt.Sprintf(
 				"func(%s)",
-				g.renderTypeTuple(ctx, t.Params()),
+				g.renderTypeTuple(ctx, t.Params(), t.Variadic()),
 			)
 		case 1:
 			return fmt.Sprintf(
 				"func(%s) %s",
-				g.renderTypeTuple(ctx, t.Params()),
+				g.renderTypeTuple(ctx, t.Params(), t.Variadic()),
 				g.renderType(ctx, t.Results().At(0).Type()),
```

Comment:
This is a minor quip about how the code currently is (not a comment on your solution here). Technically we could have an infinitely recursive situation where the function signature itself has another variadic function 😓 I think your solution is good here but this problem is a recursive one so I envision we might need to do the proper recursive solution here at some point.